logstore/store.go

```diff
@@ -0,0 +1,112 @@
+package logstore
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"sync"
+
+	"github.com/mindtastic/koda/log"
+)
+
+const (
+	logFileName          = "logstore.db"
+	defaultMaxRecordSize = 1 << 20 // 1 Megabyte
+)
+
+// Store represents a persistent, append only logbased key-value store
+type Store struct {
+	// Path of the underlying logfile
+	storagePath string
+	// Maximum allowed size ofr a single record
+	maxRecordSize int
+	// Set the sync flag to actually write to disk (using sync systemcall) after each database write access.
+	// Synchronous mode might cause dramatic performance decrease.
+	sync bool
+
+	mu sync.Mutex
+}
+
+func NewStore(storeDir string) (*Store, error) {
+	p := path.Join(storeDir, logFileName)
+
+	if _, err := os.OpenFile(p, os.O_CREATE, 0600); err != nil {
+		return nil, err
+	}
+
+	return &Store{
+		storagePath:   p,
+		maxRecordSize: defaultMaxRecordSize,
+	}, nil
+}
+
+func (s *Store) Get(key string) ([]byte, error) {
+	f, err := os.Open(s.storagePath)
```

Comment:
Each call to Get opens the file anew and creates a new scanner.

I am thinking that it would be much more efficient to implement the store using a `sync.Pool` of custom scanners that use an `io.ReadSeeker` (an `os.File` is an [`io.Seeker`](https://pkg.go.dev/os#File.Seek)).

The problem is generally that right now Get is `O(n)` instead of `O(1)`, which scales .. not great 😅